stl/inc/xutility

```diff
@@ -5563,17 +5563,36 @@ _NODISCARD constexpr auto lexicographical_compare_three_way(
 }
 #endif // __cpp_lib_concepts
 
+template <class _Ty, class _Elem>
+struct _Vector_alg_in_find_is_safe_object_pointers : false_type {};
+template <class _T1, class _T2>
+struct _Vector_alg_in_find_is_safe_object_pointers<_T1*, _T2*>
+    : disjunction<is_same<remove_cv_t<_T1>, remove_cv_t<_T2>>, is_void<_T1>, is_void<_T2>> {};

```

Comment:
This says it's for object pointers, but it's not detecting objects, nor is anything in the invocation. The syntax `T*` rejects PMFs/PMDs, but permits function pointers (as correctly noted in the comment below about finding `nullptr_t`).

So, I believe this will report `true` for `void*` and a function pointer. Those aren't supposed to be comparable, although compilers are reluctant to warn about it (see https://godbolt.org/z/x41zGYsx1 ). I think that this helper should provide behavior matching its name, activating the optimization for object pointers only. (If the plain C++ implementation happens to work for mixing `void*` and a function pointer, so be it.)

Reply:
mmh, TIL that `T*` can match a function pointer (it makes sense, but broke my expectations)